### ISSUE TITLE
The order of top-level statements and definitions now matters

### ISSUE THREAD

**Issue by skacurt** [roslynpad/roslynpad#449]
Top-level statements now only work in certain situations.

**To Reproduce**

Does not work:
```cs
class Poco  {}
var obj  = new Poco(); // err
```
Also does not work:
```cs
var obj1  = new Poco();
class Poco  {}
var obj2 = new Poco(); // err
```
Works:
```cs
var obj  = new Poco();
class Poco  {}
```
**Expected behavior**
The order of top-level statements should make no difference, as in the previous release (15.1):

An example code that is expected to work:
```cs
var poco1 = new Poco1();
class Poco1  {}
var poco2  = new Poco2();
class Poco2  {}
```

**Version**
- OS: Windows 10 Version 22H2 (OS Build 19045.2311)
- Build (or commit ID if built from source): 74d2fab6acf72792ba0328d33b7a1bb81c963815

**Comment by aelij** [comment]
Yes, this is a limitation of Roslyn - basically RP's default is to use [top level statements](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/top-level-statements). The advantage is that they don't have the limitations imposed by C# scripts - that is, they can have namespaces, public types, etc. Which is why I chose it as the new default.

You can continue using .csx files (<kbd>Ctrl+Shift+N</kbd>) and get the previous behavior.
